### PR TITLE
Forward-port and rebase of "New Capabilities plugin #3292"

### PR DESCRIPTION
**Pull request by dago**
ChangeLog: capabilities plugin: New plugin to read static platform data

### CODE REVIEW

**Comment by kwiatrox** [comment]
Thank you for the rebase @dago , but it seems that unfortunately there are still conflicts. This is probably due to recent merge of dpdk_telemetry which also introduced use of libjansson.

**Comment by dago** [comment]
@kwiatrox RIght, I fixed the conflict and repushed, please review.

File: src/daemon/filter_chain.c [hunk 1]
```diff
@@ -219,7 +219,7 @@ static int fc_config_add_match(fc_match_t **matches_head, /* {{{ */
   }
 
   if (ptr == NULL && IS_TRUE(global_option_get("AutoLoadPlugin"))) {
-    char plugin_name[NAME_MAX];
+    char plugin_name[DATA_MAX_NAME_LEN];
```
Comment (kwiatrox) [inline src/daemon/filter_chain.c]: Something went wrong, rebase probably shouldn't touch this file.

File: src/daemon/filter_chain.c [hunk 2]
```diff
@@ -306,7 +306,7 @@ static int fc_config_add_target(fc_target_t **targets_head, /* {{{ */
...
```
Comment (kwiatrox) [inline src/daemon/filter_chain.c]: Same as above.

File: src/daemon/filter_chain.c (hunk 1 above)
Comment (dago) [reply src/daemon/filter_chain.c]: Damn, this should have gone into a different branch. I'll clean this up.

File: src/daemon/filter_chain.c (hunk 2 above)
Comment (dago) [reply src/daemon/filter_chain.c]: Same.

**Comment by dago** [comment]
@kwiatrox Can you please merge #3385 which contains the extra change here? Then I rebase this PR to master and everything should be fine for merging.

**Comment by kwiatrox** [comment]
> @kwiatrox Can you please merge #3385 which contains the extra change here? Then I rebase this PR to master and everything should be fine for merging.

Yes, I can see now that this is part of other PR and should be OK. #3385 is already reviewed.

**Comment by dago** [comment]
The unassociated commit was removed, please review and merge.

**Comment by dago** [comment]
@mrunge Would you mind having a look?

**Comment by sunkuranganath** [comment]
The changes look good to me. Thanks @dago for checking through.
I used to have "merge" permissions but I dont have any now.

**Pull request merged**